core/src/test/java/org/infinispan/distribution/ZeroCapacityNodeTest.java

```diff
@@ -0,0 +1,57 @@
+package org.infinispan.distribution;
+
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfigurationBuilder;
+import org.infinispan.distribution.ch.ConsistentHash;
+import org.infinispan.distribution.ch.impl.DefaultConsistentHash;
+import org.infinispan.distribution.group.impl.PartitionerConsistentHash;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.TestingUtil;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * Test the capacity factor for lite instance
+ *
+ * @author Katia Aresti
+ * @since 9.4
+ */
+@Test(groups = "functional", testName = "distribution.ch.ZeroCapacityNodeTest")
+public class ZeroCapacityNodeTest extends MultipleCacheManagersTest {
+
+   public static final int NUM_SEGMENTS = 60;
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      // Do nothing here, create the cache managers in the test
+   }
+
+   public void testCapacityFactorForLiteInstance() {
+      GlobalConfigurationBuilder gc = new GlobalConfigurationBuilder().onZeroCapacityNode();
+
+      ConfigurationBuilder cb = new ConfigurationBuilder();
+      cb.clustering().cacheMode(CacheMode.DIST_SYNC);
+      cb.clustering().hash().numSegments(NUM_SEGMENTS);
+
+      cb.clustering().hash().capacityFactor(0.5f);
+      addClusterEnabledCacheManager(gc, cb);
+      waitForClusterToForm();
+      assertCapacityFactors(0.0f);
+   }
+
+   private void assertCapacityFactors(float... expectedCapacityFactors) {
```

Comment:
why an array?

Reply:
refactored, the other was a test I took inspiration from but not working one